### PR TITLE
[Update] add mybatis example

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Close https://github.com/pingcap/docs-cn/issues/9508

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Icemap** [comment]
issue: https://github.com/pingcap/docs-cn/issues/9508

File: develop/dev-guide-sample-application-java.md [hunk 1]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="get-code-mybatis">
+
+可以看到，JDBC 实现的代码略显冗余，需要自己管控错误处理逻辑，且不能很好的复用代码。并非最佳实践。
+
+当前开源比较流行的 Java 应用持久层框架为 [Mybatis](https://mybatis.org/mybatis-3/index.html)，我们将搭配使用 [MyBatis Generator](https://mybatis.org/generator/quickstart.html) ，以 Maven 插件的方式，对部分持久层代码进行生成。
+
+进入目录 `plain-java-mybatis` ：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:592]:
```suggestion
进入目录 `plain-java-mybatis`：
```

File: develop/dev-guide-sample-application-java.md [hunk 2]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
...
+            │   └── PlayerMapperEx.xml
+            ├── mybatis-config.xml
+            └── mybatis-generator.xml
+```
+
+其中，自动生成的文件有：
+
+- `src/main/java/com/pingcap/model/Player.java`: Player 实体类文件
+- `src/main/java/com/pingcap/model/PlayerMapper.java`: Player Mapper 的接口文件
+- `src/main/resources/mapper/PlayerMapper.xml`: Player Mapper 的 XML 映射，它是 Mybatis 用于生成对应的接口（Player Mapper 接口）的实现类的配置
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:632]:
```suggestion
- `src/main/java/com/pingcap/model/Player.java`：Player 实体类文件
- `src/main/java/com/pingcap/model/PlayerMapper.java`：Player Mapper 的接口文件
- `src/main/resources/mapper/PlayerMapper.xml`：Player Mapper 的 XML 映射，它是 Mybatis 用于生成对应的接口（Player Mapper 接口）的实现类的配置
```

File: develop/dev-guide-sample-application-java.md [hunk 3]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
...
+        <dependency>
+        <groupId>mysql</groupId>
+        <artifactId>mysql-connector-java</artifactId>
+        <version>5.1.49</version>
+        </dependency>
+    </dependencies>
+</plugin>
+```
+
+在 Maven 插件内引入后，可删除旧的生成文件后，通过命令：`mvn mybatis-generate` 生成新的文件。或者你也可以使用已经编写好的 make 命令： `make gen` 来同时删除旧文件，并生成新文件。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:744]:
```suggestion
在 Maven 插件内引入后，可删除旧的生成文件后，通过命令 `mvn mybatis-generate` 生成新的文件。或者你也可以使用已经编写好的 `make` 命令，通过 `make gen` 来同时删除旧文件，并生成新文件。
```

File: develop/dev-guide-sample-application-java.md [hunk 4]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
...
+
+> **注意：**
+>
+> `mybatis-generator.xml` 中的属性：`configuration.overwrite` 仅可控制新生成的 Java 代码文件使用覆盖方式被写入，但 XML 映射文件仍会以追加方式写入。因此，推荐在 Mybaits Generator 生成新的文件前，先删除掉旧的文件。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:748]:
```suggestion
> `mybatis-generator.xml` 中的属性 `configuration.overwrite` 仅可控制新生成的 Java 代码文件使用覆盖方式被写入，但 XML 映射文件仍会以追加方式写入。因此，推荐在 Mybaits Generator 生成新的文件前，先删除掉旧的文件。
```

File: develop/dev-guide-sample-application-java.md [hunk 5]
```diff
@@ -916,6 +1503,30 @@ mysql --host 127.0.0.1 --port 4000 -u root<src/main/resources/dbinit.sql
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="jdbc-table-init-mybatis">
+
+> 在 Gitpod Playground 中尝试 JDBC: [现在就试试](https://gitpod.io/#targetMode=plain-java-mybatis/https://github.com/pingcap-inc/tidb-example-java)
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1508]:
```suggestion
> 在 Gitpod Playground 中尝试 JDBC：[现在就试试](https://gitpod.io/#targetMode=plain-java-mybatis/https://github.com/pingcap-inc/tidb-example-java)
```

File: develop/dev-guide-sample-application-java.md [hunk 6]
```diff
@@ -1051,6 +1762,37 @@ mysql --connect-timeout 15 -u root -h tidb.e049234d.d40d1f8b.us-east-1.prod.aws.
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="run-mybatis">
+
+运行 `make`，这是以下四个操作的组合：
+
+- 创建表 (make prepare)：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1769]:
```suggestion
- 创建表 (`make prepare`)：
```

File: develop/dev-guide-sample-application-java.md [hunk 7]
```diff
@@ -1051,6 +1762,37 @@ mysql --connect-timeout 15 -u root -h tidb.e049234d.d40d1f8b.us-east-1.prod.aws.
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    mycli --host 127.0.0.1 --port 4000 -u root --no-warn < src/main/resources/dbinit.sql
+    mycli --host 127.0.0.1 --port 4000 -u root -e "TRUNCATE test.player"
+    ```
+
+- 清理并构建 (make gen)：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1778]:
```suggestion
- 清理并构建 (`make gen`)：
```

File: develop/dev-guide-sample-application-java.md [hunk 8]
```diff
@@ -1051,6 +1762,37 @@ mysql --connect-timeout 15 -u root -h tidb.e049234d.d40d1f8b.us-east-1.prod.aws.
...
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    rm -f src/main/java/com/pingcap/model/Player.java
+    rm -f src/main/java/com/pingcap/model/PlayerMapper.java
+    rm -f src/main/resources/mapper/PlayerMapper.xml
+    mvn mybatis-generator:generate
+    ```
+
+- 清理并构建 (make build)：`mvn clean package`
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1789]:
```suggestion
- 清理并构建 (`make build`)：`mvn clean package`
```

File: develop/dev-guide-sample-application-java.md [hunk 9]
```diff
@@ -1051,6 +1762,37 @@ mysql --connect-timeout 15 -u root -h tidb.e049234d.d40d1f8b.us-east-1.prod.aws.
...
+- 运行 (make run)：`java -jar target/plain-java-mybatis-0.0.1-jar-with-dependencies.jar`
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1790]:
```suggestion
- 运行 (`make run`)：`java -jar target/plain-java-mybatis-0.0.1-jar-with-dependencies.jar`
```

File: develop/dev-guide-sample-application-java.md [hunk 10]
```diff
@@ -1051,6 +1762,37 @@ mysql --connect-timeout 15 -u root -h tidb.e049234d.d40d1f8b.us-east-1.prod.aws.
...
+
+你也可以单独运行这四个 make 命令或原生命令。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1792]:
```suggestion
你也可以单独运行这四个 `make` 命令或原生命令。
```

File: develop/dev-guide-sample-application-java.md [hunk 11]
```diff
@@ -964,6 +1575,106 @@ mysqlDataSource.setPassword("123456");
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="tidb-cloud-mybatis">
+
+若你使用非本地默认集群、TiDB Cloud 或其他远程集群，更改 `mybatis-config.xml` 内关于 dataSource.url、dataSource.username、dataSource.password 的参数：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1580]:
```suggestion
若你使用非本地默认集群、TiDB Cloud 或其他远程集群，更改 `mybatis-config.xml` 内关于 `dataSource.url`、`dataSource.username`、`dataSource.password` 的参数：
```

File: develop/dev-guide-sample-application-java.md [hunk 12]
```diff
@@ -916,6 +1503,30 @@ mysql --host 127.0.0.1 --port 4000 -u root<src/main/resources/dbinit.sql
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="jdbc-table-init-mybatis">
+
+> 在 Gitpod Playground 中尝试 JDBC: [现在就试试](https://gitpod.io/#targetMode=plain-java-mybatis/https://github.com/pingcap-inc/tidb-example-java)
+
+使用 JDBC 时，需手动初始化数据库表，若你本地已经安装了 `mysql-client`，且使用本地集群，可直接在 `plain-java-mybatis` 目录下运行：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1510]:
```suggestion
使用 JDBC 时，需手动初始化数据库表。若你本地已经安装了 `mycli`，且使用本地集群，可直接在 `plain-java-mybatis` 目录下运行：
```

File: develop/dev-guide-sample-application-java.md [hunk 13]
```diff
@@ -916,6 +1503,30 @@ mysql --host 127.0.0.1 --port 4000 -u root<src/main/resources/dbinit.sql
...
+
+或直接执行：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+mycli --host 127.0.0.1 --port 4000 -u root --no-warn < src/main/resources/dbinit.sql
+```
+
+若你不使用本地集群，或未安装 **mysql-client**，请直接登录你的集群，并运行 `src/main/resources/dbinit.sql` 文件内的 SQL 语句。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1526]:
```suggestion
若你不使用本地集群，或未安装 mycli，请直接登录你的集群，并运行 `src/main/resources/dbinit.sql` 文件内的 SQL 语句。
```

File: develop/dev-guide-sample-application-java.md [hunk 14]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
...
+  </select>
+
+  <select id="count" resultType="java.lang.Integer">
+    select count(*) from player
+  </select>
+
+</mapper>
+```
+
+`PlayerDAO.java` 是程序用来管理数据对象的类。其中 `DAO` 是 [Data Access Object](https://en.wikipedia.org/wiki/Data_access_object) 的缩写。在其中定义了一系列数据的操作方法，用来对提供数据的写入能力。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:981]:
```suggestion
`PlayerDAO.java` 是程序用来管理数据对象的类。其中 `DAO` 是 [Data Access Object](https://en.wikipedia.org/wiki/Data_access_object) 的缩写。在其中定义了一系列数据的操作方法，用于数据的写入。
```

File: develop/dev-guide-sample-application-java.md [hunk 15]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="get-code-mybatis">
+
+可以看到，JDBC 实现的代码略显冗余，需要自己管控错误处理逻辑，且不能很好的复用代码。并非最佳实践。
+
+当前开源比较流行的 Java 应用持久层框架为 [Mybatis](https://mybatis.org/mybatis-3/index.html)，我们将搭配使用 [MyBatis Generator](https://mybatis.org/generator/quickstart.html) ，以 Maven 插件的方式，对部分持久层代码进行生成。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:590]:
```suggestion
当前开源比较流行的 Java 应用持久层框架为 [Mybatis](https://mybatis.org/mybatis-3/index.html)，本文将搭配使用 [MyBatis Generator](https://mybatis.org/generator/quickstart.html) ，以 Maven 插件的方式，对部分持久层代码进行生成。
```

File: develop/dev-guide-sample-application-java.md [hunk 16]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
...
+            └── mybatis-generator.xml
+```
+
+其中，自动生成的文件有：
+
+- `src/main/java/com/pingcap/model/Player.java`：Player 实体类文件
+- `src/main/java/com/pingcap/model/PlayerMapper.java`：Player Mapper 的接口文件
+- `src/main/resources/mapper/PlayerMapper.xml`：Player Mapper 的 XML 映射，它是 Mybatis 用于生成对应的接口（Player Mapper 接口）的实现类的配置
+
+这些文件的生成策略被写在了 `mybatis-generator.xml` 配置文件内，它是 [Mybatis Generator](https://mybatis.org/generator/quickstart.html) 的配置文件，已在配置文件中进行逐行解释了使用方法：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:634]:
```suggestion
这些文件的生成策略被写在了 `mybatis-generator.xml` 配置文件内，它是 [Mybatis Generator](https://mybatis.org/generator/quickstart.html) 的配置文件，下面配置文件中添加了使用方法的说明：
```

File: develop/dev-guide-sample-application-java.md [hunk 17]
```diff
@@ -916,6 +1503,30 @@ mysql --host 127.0.0.1 --port 4000 -u root<src/main/resources/dbinit.sql
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="jdbc-table-init-mybatis">
+
+> 在 Gitpod Playground 中尝试 JDBC：[现在就试试](https://gitpod.io/#targetMode=plain-java-mybatis/https://github.com/pingcap-inc/tidb-example-java)
+
+使用 JDBC 时，需手动初始化数据库表。若你本地已经安装了 `mysql-client`，且使用本地集群，可直接在 `plain-java-mybatis` 目录下运行：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1510]:
```suggestion
使用 JDBC 时，需手动初始化数据库表。若你本地已经安装了 `mysql-client`，且使用本地集群，可直接在 `plain-java-mybatis` 目录下通过 `make prepare` 运行：
```

File: develop/dev-guide-sample-application-java.md [hunk 18]
```diff
@@ -916,6 +1503,30 @@ mysql --host 127.0.0.1 --port 4000 -u root<src/main/resources/dbinit.sql
...
+
+或直接执行：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+mysql --host 127.0.0.1 --port 4000 -u root < src/main/resources/dbinit.sql
+```
+
+若你不使用本地集群，或未安装 mysql-client，请直接登录你的集群，并运行 `src/main/resources/dbinit.sql` 文件内的 SQL 语句。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:1526]:
```suggestion
若你不使用本地集群，或未安装 `mysql-client`，请直接登录你的集群，并运行 `src/main/resources/dbinit.sql` 文件内的 SQL 语句。
```

File: develop/dev-guide-sample-application-java.md [hunk 19]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="get-code-mybatis">
+
+可以看到，JDBC 实现的代码略显冗余，需要自己管控错误处理逻辑，且不能很好的复用代码。并非最佳实践。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:588]:
```suggestion
可以看到，JDBC 实现的代码略显冗余，需要自己管控错误处理逻辑，且不能很好的复用代码，并非最佳实践。
```

File: develop/dev-guide-sample-application-java.md [hunk 20]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="get-code-mybatis">
+
+可以看到，JDBC 实现的代码略显冗余，需要自己管控错误处理逻辑，且不能很好的复用代码，并非最佳实践。
+
+当前开源比较流行的 Java 应用持久层框架为 [Mybatis](https://mybatis.org/mybatis-3/index.html)，本文将搭配使用 [MyBatis Generator](https://mybatis.org/generator/quickstart.html) ，以 Maven 插件的方式，对部分持久层代码进行生成。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-java.md:590]:
```suggestion
当前开源比较流行的 Java 应用持久层框架为 [Mybatis](https://mybatis.org/mybatis-3/index.html)，本文将搭配使用 [MyBatis Generator](https://mybatis.org/generator/quickstart.html)，以 Maven 插件的方式，对部分持久层代码进行生成。
```

**Review by Oreoxmt — APPROVED** [review]

File: develop/dev-guide-sample-application-java.md [hunk 21]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
 
 </div>
 
+<div label="使用 Mybatis（推荐）" href="get-code-mybatis">
+
+可以看到，JDBC 实现的代码略显冗余，需要自己管控错误处理逻辑，且不能很好的复用代码，并非最佳实践。
+
+当前开源比较流行的 Java 应用持久层框架为 [Mybatis](https://mybatis.org/mybatis-3/index.html)，本文将搭配使用 [MyBatis Generator](https://mybatis.org/generator/quickstart.html)，以 Maven 插件的方式，对部分持久层代码进行生成。
```
Comment (Mini256) [inline develop/dev-guide-sample-application-java.md:590]:
```suggestion
[Mybatis](https://mybatis.org/mybatis-3/index.html) 是当前比较流行的开源 Java 应用持久层框架，本文将以 Maven 插件的方式使用 [MyBatis Generator](https://mybatis.org/generator/quickstart.html) 生成部分持久层代码。
```

File: develop/dev-guide-sample-application-java.md [hunk 22]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
...
+            │   └── PlayerMapperEx.xml
+            ├── mybatis-config.xml
+            └── mybatis-generator.xml
+```
+
+其中，自动生成的文件有：
+
+- `src/main/java/com/pingcap/model/Player.java`：Player 实体类文件
+- `src/main/java/com/pingcap/model/PlayerMapper.java`：Player Mapper 的接口文件
+- `src/main/resources/mapper/PlayerMapper.xml`：Player Mapper 的 XML 映射，它是 Mybatis 用于生成对应的接口（Player Mapper 接口）的实现类的配置
```
Comment (Mini256) [inline develop/dev-guide-sample-application-java.md:632]: 前面已经特指 Player Mapper 接口了

```suggestion
- `src/main/resources/mapper/PlayerMapper.xml`：Player Mapper 的 XML 映射，它是 Mybatis 用于生成 Player Mapper 接口的实现类的配置
```

File: develop/dev-guide-sample-application-java.md [hunk 23]
```diff
@@ -583,6 +583,593 @@ public class JDBCExample
...
+        -->
+        <table tableName="player" domainObjectName="Player"
+               enableCountByExample="false" enableUpdateByExample="false"
+               enableDeleteByExample="false" enableSelectByExample="false"
+               selectByExampleQueryId="false"/>
+    </context>
+</generatorConfiguration>
+```
+
+`mybatis-generator.xml` 在 `pom.xml` 中，以 Maven 插件的方式被引入：
```
Comment (Mini256) [inline develop/dev-guide-sample-application-java.md:720]:
```suggestion
`mybatis-generator.xml` 在 `pom.xml` 中，以 `mybatis-generator-maven-plugin` 插件配置的方式被引入：
```

File: develop/dev-guide-sample-application-java.md [hunk 24]
```diff
@@ -964,6 +1575,106 @@ mysqlDataSource.setPassword("123456");
...
+
+若你设定的密码为 `123456`，而且从 TiDB Cloud 得到的连接字符串为：
+
+```
+mysql --connect-timeout 15 -u root -h tidb.e049234d.d40d1f8b.us-east-1.prod.aws.tidbcloud.com -P 4000 -p
+```
+
+那么此处应将配置文件更改为：
+
+{{< copyable "" >}}
```
Comment (Mini256) [inline develop/dev-guide-sample-application-java.md:1633]: @shczhen, 我们支持行高亮吗？

@Icemap  或者只留下改的 datasource配置。

Comment (Icemap) [reply develop/dev-guide-sample-application-java.md:1633]: Done

**Review by TomShawn — APPROVED** [review]

**Pull request merged**